stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -1,7 +1,7 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 # Symmetric and Hermitian matrices
-struct Symmetric{T,S<:AbstractMatrix{T}} <: AbstractMatrix{T}
+struct Symmetric{T,S<:AbstractMatrix{<:T}} <: AbstractMatrix{T}
```

Comment:
Could you explain this change here?

Reply:
The type of `transpose(A[i, j])` may not be the same as that of `A[i, j]`, so in order to not lie about the return type of `getindex(A, i, j)`, `eltype(A)` may need to be widened with respect to `eltype(A.data)`.

Reply:
I’m not sure this is quite right. The `Adjoint` stuff might not even match this signature. IIRC, for `Transpose` and `Adjoint` we leave this free and check it later (in constructors, getindex, etc).

Reply:
I'm confused, maybe I'm misunderstanding you. If `A` is the argument to the `Symmetric` outer constructor, then `T == Union{eltype(A),promote_op(transpose, eltype(A))}` should always be at least `eltype(A) == eltype(S)` (so that `eltype(S) <: T`), right?

Do you mean that it would be possible to call the inner constructor with bad parameters? That I agree with. Could make the current outer constructor an inner constructor.

Reply:
I second @tkoolen's explanation and share his confusion.

Reply:
This definition now clicks on this end. Thanks for the explanations! :)